lib/preload.ts

```diff
@@ -564,53 +563,65 @@ export class Preloader extends EventEmitter {
 	}
 
 	async registry(
-		registryUrl,
-		endpoint,
-		registryToken,
-		headers,
-		decodeJson,
-		followRedirect,
-		encoding?,
+		registryUrl: string,
+		endpoint: string,
+		registryToken: string,
+		headers: Record<string, string>,
+		decodeJson: boolean,
+		followRedirect: boolean,
 	) {
-		headers = { ...headers };
-		headers['Authorization'] = `Bearer ${registryToken}`;
-		return await request({
+		return await this.balena.request.send({
 			url: `https://${registryUrl}${endpoint}`,
-			headers,
+			headers: {
+				...headers,
+				Authorization: `Bearer ${registryToken}`,
+			},
 			json: decodeJson,
-			simple: false,
-			resolveWithFullResponse: true,
+			...(decodeJson && { responseFormat: 'blob' }),
```

Comment:
@thgreasi shouldn't this be `...(decodeJson && { responseFormat: 'blob' })` ? My understanding was if we have decodeJson we want JSON output, otherwise, we want blob ?